src/rules/a11y-tooltip-non-interactive-trigger.js

```diff
@@ -0,0 +1,184 @@
+  const isAnchorInteractive = typeof href === 'string' && href !== ''
+  return isAnchorInteractive
+}
+
+const isInputTag = el => {
+  return (
+    getJSXOpeningElementName(el.openingElement) === 'input' ||
+    getJSXOpeningElementName(el.openingElement).toLowerCase() === 'textinput'
+  )
+}
+
+const isInteractiveInput = child => {
+  const hasHiddenType =
+    getJSXOpeningElementAttribute(child.openingElement, 'type') &&
+    getJSXOpeningElementAttribute(child.openingElement, 'type').value.value === 'hidden'
+  return !hasHiddenType
+}
+
+const isOtherThanAnchorOrInput = el => {
+  return !isAnchorTag(el) && !isInputTag(el)
+}
+
+const getAllChildren = node => {
+  if (Array.isArray(node.children)) {
+    return node.children
+      .filter(child => {
+        return child.type === 'JSXElement'
+      })
+      .flatMap(child => {
+        return [child, ...getAllChildren(child)]
+      })
+  }
+  return []
+}
+
+const checks = [
+  {
+    id: 'anchorTagWithoutHref',
+    filter: jsxElement => isAnchorTag(jsxElement),
+    check: isInteractiveAnchor
+  },
+  {
+    id: 'hiddenInput',
+    filter: jsxElement => isInputTag(jsxElement),
+    check: isInteractiveInput
+  },
+  {
+    id: 'nonInteractiveTrigger',
+    filter: jsxElement => isOtherThanAnchorOrInput(jsxElement),
+    check: isInteractive
+  }
+]
+
+const checkTriggerElement = jsxNode => {
+  const elements = [...getAllChildren(jsxNode)]
+  const hasInteractiveElement = elements.find(element => {
+    if (
+      getJSXOpeningElementName(element.openingElement) === 'a' ||
+      getJSXOpeningElementName(element.openingElement) === 'Link'
+    ) {
+      return isInteractiveAnchor(element)
+    }
+    if (
+      getJSXOpeningElementName(element.openingElement) === 'input' ||
+      getJSXOpeningElementName(element.openingElement) === 'TextInput'
+    ) {
+      return isInteractiveInput(element)
+    } else {
+      return isInteractive(element)
+    }
+  })
+
+  // If the tooltip has interactive elements, return.
+  if (hasInteractiveElement) return
+
+  const errors = new Set()
+
+  for (const element of elements) {
+    for (const check of checks) {
+      if (!check.filter(element)) {
+        continue
+      }
+
+      if (!check.check(element)) {
+        errors.add(check.id)
+      }
+    }
+  }
+  // check the specificity of the errors. If there are multiple errors, only return the most specific one.
+  if (errors.size > 1) {
+    if (errors.has('anchorTagWithoutHref')) {
+      errors.delete('nonInteractiveTrigger')
+    }
+    if (errors.has('hiddenInput')) {
+      errors.delete('nonInteractiveTrigger')
+    }
+  }
+
+  return errors
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      url:
+        'https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/no-noninteractive-element-interactions.md',
+      description: 'Non-interactive elements should not be assigned mouse or keyboard event listeners.'
```

Comment:
Should this description be more specific to Tooltip?

Reply:
Oh I think I copied this from somewhere to change it later and forgot! Thanks for the catch. I'll update it

Reply:
I ended up removing this because I don't see it in other rules and I don't also know where it would appear 🤔 Let me know what you think